### PR TITLE
DOCSP-40803: Cursor guide

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-40803
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/cpp-driver/DOCSP-40803-cursor-guide/read/cursor/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/read/cursor.txt [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+Overview
+--------
+
+In this guide, you can learn how to access data from a **cursor** by using the
+{+driver-short+}.
+
+A cursor is a mechanism that returns the results of a read operation in iterable
+batches. Because a cursor holds only a subset of documents at any given time,
+cursors reduce both memory consumption and network bandwidth usage.
```
Comment (shuangela) [inline source/read/cursor.txt:28]: Q: cursors reduce memory consumption and network bandwidth usage compared to what? A regular read operation?

File: source/read/cursor.txt [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+   .. input:: /includes/read/cursor.cpp
+      :start-after: start-tailable
+      :end-before: end-tailable
+      :language: cpp
+      :dedent:
+
+   .. output::
+
+      { "_id" : { "$oid" : "..." }, "name" : "cauliflower" }
```
Comment (shuangela) [inline source/read/cursor.txt:134]: Q: The copy talks about retrieving three documents, but the output only shows two documents.

File: source/read/cursor.txt [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+      { "_id" : { "$oid" : "..." }, ... "name" : "Dunkin' Donuts", "restaurant_id" : "40379573" }
+      { "_id" : { "$oid" : "..." }, ... "name" : "Dunkin' Donuts", "restaurant_id" : "40363098" }
+      { "_id" : { "$oid" : "..." }, ... "name" : "Dunkin' Donuts", "restaurant_id" : "40395071" }
+      ...
+
+Retrieve Documents Individually
+-------------------------------
+
+To retrieve documents from a cursor individually, call the ``begin()`` method on
```
Comment (shuangela) [inline source/read/cursor.txt:80]:
```suggestion
To retrieve an individual document from a cursor, call the ``begin()`` method on
```

**Review by shuangela — CHANGES_REQUESTED** [review]
Really good work! Couple of small comments.

File: source/read/cursor.txt (hunk 1 above)
Comment (norareidy) [reply source/read/cursor.txt:28]: Compared to returning all the documents at once - I'll clarify that

File: source/read/cursor.txt (hunk 2 above)
Comment (norareidy) [reply source/read/cursor.txt:134]: Line 137 ("if you insert another document ...") kinda explains that, but let me know if it should be clearer. Basically there are two documents in the collection, so the cursor will retrieve two, but it's a tailable cursor so it stays open even though the original cursor is exhausted. Then if you insert a third document, the cursor will also store that document.

Comment (shuangela) [reply source/read/cursor.txt:135]: Ah, ok, I see. Maybe edit the "continues to check the cursor" line to be something like:

"keeps the cursor open to check for more documents until it retrieves three documents"

**Review by shuangela — APPROVED** [review]
LGTM! Small comment.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**